x/ecocredit/server/marketplace/buy.go

```diff
@@ -10,96 +10,21 @@ import (
 	"github.com/regen-network/regen-ledger/x/ecocredit"
 	"github.com/regen-network/regen-ledger/x/ecocredit/core"
 	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
-	"github.com/regen-network/regen-ledger/x/ecocredit/server/utils"
 
 	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-// Buy allows users to purchase credits by either directly specifying a sell order, or
-// defining a set of filters with attributes to match against.
-//
-// Currently, only the former is supported. Calls to this function with anything other than
-// MsgBuy_Order_Selection_SellOrderId will fail.
+// Buy creates a buy order to be used in the order-book.
```

Comment:
Both direct and filtered buy orders will still be used alongside the orderbook as far as I understand.

```suggestion
// Buy creates a buy order that matches sell orders based on filter criteria.
```